cassiopeia/solver/VanillaGreedySolver.py

```diff
@@ -78,33 +78,56 @@ def perform_split(
         Returns:
             A tuple of lists, representing the left and right partitions
         """
-        freq = 0
-        char = 0
-        state = ""
-        for i in mutation_frequencies:
-            for j in mutation_frequencies[i]:
-                if j != self.missing_char and j != "0":
+        best_frequency = 0
+        chosen_character = 0
+        chosen_state = ""
+        for character in mutation_frequencies:
+            for state in mutation_frequencies[character]:
+                if state != self.missing_char and state != "0":
                     # Avoid splitting on mutations shared by all samples
                     if (
-                        mutation_frequencies[i][j] > freq
-                        and mutation_frequencies[i][j]
+                        mutation_frequencies[character][state]
                         < len(samples)
-                        - mutation_frequencies[i][self.missing_char]
+                        - mutation_frequencies[character][self.missing_char]
                     ):
-                        char, state = i, j
-                        freq = mutation_frequencies[i][j]
-
-        if state == "":
+                        if self.priors:
+                            if (
+                                mutation_frequencies[character][state]
+                                * self.priors[character][state]
+                                > best_frequency
+                            ):
+                                chosen_character, chosen_state = (
+                                    character,
+                                    state,
+                                )
+                                best_frequency = (
+                                    mutation_frequencies[character][state]
+                                    * self.priors[character][state]
+                                )
+                        else:
+                            if (
+                                mutation_frequencies[character][state]
+                                > best_frequency
+                            ):
+                                chosen_character, chosen_state = (
+                                    character,
+                                    state,
+                                )
+                                best_frequency = mutation_frequencies[
+                                    character
+                                ][state]
```

Comment:
I think this definitely needs to be changed. Should be `-nlog(p)` as we had in the original paper.